src/controllers/controllers.ts

```diff
@@ -4,16 +4,28 @@ import { UserService } from '../services/userService';
 import { Controller } from '../models/controller';
 import { GroupService } from '../services/groupService';
 import { HttpCode } from '../models/interfaces';
+import { addPagination } from '../helpers/pagination';
 
 const usersController = new Controller(new UserService());
 const groupsController = new Controller(new GroupService());
 const userGroupService = new UserGroupService(new GroupService(), new UserService());
 
+const getAllUsersWithAutoSuggestion = async (req: Request, res: Response) => {
+    const { limit, loginSubstring, skip } = req.query as { [param: string]: string | undefined };
+
+    if (limit && loginSubstring) {
+        const items = await (usersController.service as UserService).getAutoSuggestions(limit, loginSubstring);
```

Comment:
Makes sense to add a generic TService type to the Controller or use `userService` directly to avoid type casting